dashboard/app/controllers/sessions_controller.rb

```diff
@@ -6,6 +6,12 @@ class SessionsController < Devise::SessionsController
   def new
     session[:return_to] ||= params[:return_to]
     @already_hoc_registered = params[:already_hoc_registered]
+    @hide_oauth_sidebar = false
+    if params[:providerNotLinked]
+      flash.now[:alert] = I18n.t 'auth.not_linked', provider: I18n.t("auth.#{params[:providerNotLinked]}")
```

Comment:
I think the motivation is that in other languages, we'll need translations when we refer to accounts. For instance, in https://github.com/code-dot-org/code-dot-org/blob/f614da2012a10b224c0e22178125bba8a066af5e/dashboard/config/locales/es-ES.yml#L83-L86 instead of "Google Account" we have "Cuenta de Google".

You're absolutely right that this should be documented.